src/webgpu/web_platform/external_texture/video.spec.ts

```diff
@@ -27,46 +27,58 @@ const kVideoInfo = /* prettier-ignore */ makeTable(
+  'four-colors-vp8-bt601.webm'  : [    'REC601',         'video/webm; codecs=vp8'],
+  'four-colors-theora-bt601.ogv': [    'REC601',       'video/ogg; codecs=theora'],
+  'four-colors-h264-bt601.mp4'  : [    'REC601',  'video/mp4; codecs=avc1.4d400c'],
+  'four-colors-vp9-bt601.webm'  : [    'REC601',         'video/webm; codecs=vp9'],
+  'four-colors-vp9-bt709.webm'  : [    'REC709',         'video/webm; codecs=vp9'],
+  'four-colors-vp9-bt2020.webm' : [   'REC2020',         'video/webm; codecs=vp9']
 } as const);
 type VideoName = keyof typeof kVideoInfo;
 type VideoInfo = valueof<typeof kVideoInfo>;
 
+// The expected color values were calculated based on:
+// https://chromium-review.googlesource.com/c/chromium/src/+/4232318
+const kBt601Red = new Uint8Array([248, 36, 0, 255]);
+const kBt601Green = new Uint8Array([64, 252, 0, 255]);
+const kBt601Blue = new Uint8Array([26, 35, 255, 255]);
+const kBt601Yellow = new Uint8Array([254, 253, 0, 255]);
+
 const kVideoExpectations = [
   {
-    videoName: 'red-green.webmvp8.webm',
-    _redExpectation: new Uint8Array([0xf8, 0x24, 0x00, 0xff]),
-    _greenExpectation: new Uint8Array([0x3f, 0xfb, 0x00, 0xff]),
-  },
-  {
-    videoName: 'red-green.theora.ogv',
-    _redExpectation: new Uint8Array([0xf8, 0x24, 0x00, 0xff]),
-    _greenExpectation: new Uint8Array([0x3f, 0xfb, 0x00, 0xff]),
+    videoName: 'four-colors-vp8-bt601.webm',
+    _redExpectation: kBt601Red,
+    _greenExpectation: kBt601Green,
+    _blueExpectation: kBt601Blue,
+    _yellowExpectation: kBt601Yellow,
   },
   {
-    videoName: 'red-green.mp4',
-    _redExpectation: new Uint8Array([0xf8, 0x24, 0x00, 0xff]),
-    _greenExpectation: new Uint8Array([0x3f, 0xfb, 0x00, 0xff]),
+    videoName: 'four-colors-theora-bt601.ogv',
+    _redExpectation: kBt601Red,
+    _greenExpectation: kBt601Green,
+    _blueExpectation: kBt601Blue,
+    _yellowExpectation: kBt601Yellow,
   },
   {
-    videoName: 'red-green.bt601.vp9.webm',
-    _redExpectation: new Uint8Array([0xf8, 0x24, 0x00, 0xff]),
-    _greenExpectation: new Uint8Array([0x3f, 0xfb, 0x00, 0xff]),
+    videoName: 'four-colors-h264-bt601.mp4',
+    _redExpectation: kBt601Red,
+    _greenExpectation: kBt601Green,
+    _blueExpectation: kBt601Blue,
+    _yellowExpectation: kBt601Yellow,
   },
   {
-    videoName: 'red-green.bt709.vp9.webm',
-    _redExpectation: new Uint8Array([0xff, 0x00, 0x00, 0xff]),
-    _greenExpectation: new Uint8Array([0x00, 0xff, 0x00, 0xff]),
+    videoName: 'four-colors-vp9-bt601.webm',
+    _redExpectation: kBt601Red,
+    _greenExpectation: kBt601Green,
+    _blueExpectation: kBt601Blue,
+    _yellowExpectation: kBt601Yellow,
   },
   {
-    videoName: 'red-green.bt2020.vp9.webm',
-    _redExpectation: new Uint8Array([0xff, 0x00, 0x00, 0xff]),
-    _greenExpectation: new Uint8Array([0x00, 0xff, 0x00, 0xff]),
+    videoName: 'four-colors-vp9-bt709.webm',
+    _redExpectation: new Uint8Array([255, 0, 1, 255]),
```

Comment:
I disagree with the expectations here. When converting BT.709 primaries to sRGB, the colors should remain "pure". Because each channel is either '1' or '0', converting according to the gamma curve either way should not have any effect. BT.709 uses the exact same gamut as sRGB, so any "pure" source color will still be "pure" after conversion.

I think this is a case where we need to allow a small tolerance.

Reply:
I introduced this (1 / 255) diff for "unorm->float" precision loss as revealed in https://chromium-review.googlesource.com/c/chromium/src/+/4232318/4/ui/gfx/color_transform_unittest.cc#227.
However I agree that it's kind of distracting. Actually the new test cases can still pass without this. So let me revert it.